src/mailroom.py

```diff
@@ -0,0 +1,122 @@
+# _*_ coding: utf-8 _*_
+import sys
+import traceback
+
+
+try:
+    input = raw_input
+except NameError:
+    pass
+
+
+DONORS = {
+    'kevin': [23, 34, 1000, 2849],
+    'bob': [234, 888, 9023],
+    'josh': [6, 1, 9, 5000, 9, 1, 2, 3],
+    'alice': [3, 1]
+}
+
+
+def interface():
+    '''Prompts user for input'''
+    response = input(u"Press 1 to send a thank you note\n"
+                     u"Press 2 to create a report\n"
+                     u"Type exit to leave the program")
+    while (response != '1') and (response != '2') and (response != 'exit'):
+        print(u"Please enter 1 or 2")
+        response = input(u"Press 1 to send a thank you note\n"
+                         u"Press 2 to create a report ")
+    if response == '1':
+        name = input(u"Enter a donor\'s full name or enter the "
+                     u"word 'list' to see a list of donors ")
+        name = name.lower()
+        while name == "list":
+            for key in DONORS:
+                print(key)
+            name = input(u"Enter a donor\'s full name or enter the "
+                         u"word 'list' to see a list of donors ")
+            name = name.lower()
+        donation = input(u"Enter the donation amount for {0}".format(name))
+        while not donation.isnumeric():
+            print("Please enter the donation using numbers.")
+            donation = input(u"Enter the donation amount for {0}".format(name))
+        handle_name(name, donation, DONORS)
+        handle_donation(name, donation, DONORS)
+        compose_email(name, donation)
+    elif response == '2':
+        create_report()
+    else:
+        traceback.print_exc(file=sys.stdout)
+        sys.exit(0)
+
+
+def handle_name(name, donation, DONORS):
+    if name not in DONORS:
+        DONORS[name] = []
+    return DONORS
```

Comment:
How is this different from `dict.setdefault(name, [])`?  If it isn't different, why do it?  Can it be combined with the `handle_donation` function below?